lalrpop/src/lr1/codegen/mod.rs

```diff
@@ -1,3 +1,5 @@
+// In code generation we may output format strings that will be used by the generated code
+#![allow(clippy::literal_string_with_formatting_args)]
```

Comment:
Thoughts on using `#![expect(...)]` over allow here in case they end up fixing this false positive?

Reply:
I supposed I'm not opposed to `expect`, although I'm not really sure what clippy can do in this case.  There was [some discussion](https://github.com/rust-lang/rust-clippy/pull/13410#issuecomment-2372387787) in the PR adding this lint about false positives in crates that manipulate format strings, and the suggestion was that those crates should allow the lint globally, which makes sense.

We're in a pretty similar boat doing code generation I think.  I'm not sure how clippy would ever know that our use is legitimate.  We write a string out to a file and it contains format characters.  The fact that that file happens to contain valid rust and is later compiled is far outside of what clippy can know.